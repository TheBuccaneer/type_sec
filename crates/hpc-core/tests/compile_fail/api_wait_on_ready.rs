use hpc_core::*;

fn main() {
    let ctx = Context::create_context().unwrap();
    let queue = ctx.create_queue().unwrap();

    // Buffer ist bereits Ready
    let buf = ctx
        .create_buffer::<u8>(16).unwrap()
        .enqueue_write(&queue, &[0u8; 16]).unwrap(); // → Ready

    // ❌ Das sollte NICHT kompilieren
    // Ready-Buffer hat keine wait()-Methode!
    let _illegal = buf.wait(); 
    //                ^^^^ should not exist on Ready state
}