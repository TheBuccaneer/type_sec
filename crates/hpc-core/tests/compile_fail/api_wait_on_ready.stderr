error[E0599]: no method named `wait` found for struct `DeviceBuffer` in the current scope
  --> tests/compile_fail/api_wait_on_ready.rs:14:24
   |
14 |     let _illegal = buf.wait();
   |                        ^^^^ method not found in `DeviceBuffer<'_, u8, hpc_core::Ready>`
